enterprise/api/v1/views/enterprise_subsidy_fulfillment.py

```diff
@@ -94,47 +95,62 @@ class EnterpriseSubsidyFulfillmentViewSet(EnterpriseWrapperApiViewSet):
+        elif self.action == 'cancel_enrollment':
+            return ['enterprise.can_enroll_learners']
+        return []
+
+    def get_permission_object(self):
+        """
+        Depending on the requested view, returns a record identifier to check perms against.
+        """
+        if self.request.user.is_staff:
+            return None
 
-        # Get learner credit enrollments under the supplied fulfillment source uuid.
-        learner_credit_enrollments = models.LearnerCreditEnterpriseCourseEnrollment.objects.filter(
-            uuid=fulfillment_source_uuid
-        )
+        if self.action in ('retrieve', 'cancel_enrollment'):
+            enrollment_record = self.requested_fulfillment_source.enterprise_course_enrollment
+            return str(enrollment_record.enterprise_customer_user.enterprise_customer.uuid)
+        else:
+            return None
 
-        # Filters to match fulfillment enrollments' and entitlements' enterprise customer uuid to the requesting
-        # user's enterprise customer uuid.
-        subsidy_fulfillment_filter = Q(
-            enterprise_course_enrollment__enterprise_customer_user__enterprise_customer__uuid=enterprise_customer_uuid
-        )
-        subsidy_fulfillment_filter |= Q(
-            enterprise_course_entitlement__enterprise_customer_user__enterprise_customer__uuid=enterprise_customer_uuid
-        )
+    @cached_property
+    def requested_fulfillment_source(self):
+        """
+        Computes and caches the requested fulfillment source record for this request.
+        """
+        return self._get_single_fulfillment_by_uuid()
 
-        # If the requester isn't staff, apply the filters
-        if not self.request.user.is_staff:
-            learner_credit_enrollments = learner_credit_enrollments.filter(subsidy_fulfillment_filter)
+    def _get_single_fulfillment_by_uuid(self):
+        """
+        Returns a single LearnerCreditCourseEnrollment or LicensedCourseEnrollment associated
+        with the requested ``fulfillment_source_uuid``.
 
-        # Return if we get any hits
-        if learner_credit_enrollments:
-            return learner_credit_enrollments
+        Raises: a 404 if no such record can be found.
+        """
+        fulfillment_source_uuid = self.kwargs.get('fulfillment_source_uuid')
 
-        # Get licensed enrollments under the supplied fulfillment source uuid and repeat the same process.
-        licensed_enrollments = models.LicensedEnterpriseCourseEnrollment.objects.filter(
-            uuid=fulfillment_source_uuid
-        )
-        if not self.request.user.is_staff:
-            licensed_enrollments = licensed_enrollments.filter(subsidy_fulfillment_filter)
+        # Get learner credit fulfillments under the supplied fulfillment source uuid.
+        learner_credit_enrollment = models.LearnerCreditEnterpriseCourseEnrollment.objects.filter(
+            uuid=fulfillment_source_uuid,
+        ).first()
```

Comment:
Is this a candidate for .get() instead of .filter().first()?  The `uuid` field has a unique constraint.